### PR TITLE
Validate bindata

### PR DESCRIPTION
**Pull request by kargakis**
Brings back the script that validates generated code that was used once and hooks it up in CI.

Fixes https://github.com/Azure/acs-engine/issues/2710

@jim-minter @jackfrancis @CecileRobertMichon ptal

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,6 +44,10 @@ all: build
 dev:
 	$(DEV_ENV_CMD_IT) bash
 
+.PHONY: validate
```
[the remark this answers is not in the document]
Comment (kargakis) [reply Makefile]: I made it generic in the hope that future source validation unrelated to generation could be dropped here. Renamed to `validate-generated` for now.

**Comment by CecileRobertMichon** [comment]
@kargakis is this ready for merge?

**Comment by kargakis** [comment]
Yes, it's ready

@jim-minter fyi

**Comment by jim-minter** [comment]
+1 thanks @kargakis

**Comment by jackfrancis** [comment]
@JiangtianLi do we want to apply these same validations go go-bindata code that MS manages?

**Comment by JiangtianLi** [comment]
@jackfrancis I don't have a context here, what validation does it do on go-bindata code?

**Pull request merged**